subxt/src/tx/tx_progress.rs

```diff
@@ -370,3 +403,113 @@ impl<T: Config, C: OnlineClientT<T>> TxInBlock<T, C> {
         ))
     }
 }
+
+#[cfg(test)]
+mod test {
+    use std::pin::Pin;
+
+    use futures::Stream;
+    use primitive_types::H256;
+    use serde::Serialize;
+
+    use crate::{
+        client::{OfflineClientT, OnlineClientT},
+        config::polkadot::PolkadotConfig,
+        error::RpcError,
+        rpc::{types::SubstrateTxStatus, RpcSubscription, Subscription},
+        tx::TxProgress,
+        Error,
+    };
+
+    use serde_json::value::RawValue;
+
+    #[derive(Clone, Debug)]
+    struct MockClient;
+
+    impl OfflineClientT<PolkadotConfig> for MockClient {
+        fn metadata(&self) -> crate::Metadata {
+            panic!("just a mock impl to satisfy trait bounds")
+        }
+
+        fn genesis_hash(&self) -> <PolkadotConfig as crate::Config>::Hash {
+            panic!("just a mock impl to satisfy trait bounds")
+        }
+
+        fn runtime_version(&self) -> crate::rpc::types::RuntimeVersion {
+            panic!("just a mock impl to satisfy trait bounds")
+        }
+    }
+
+    impl OnlineClientT<PolkadotConfig> for MockClient {
+        fn rpc(&self) -> &crate::rpc::Rpc<PolkadotConfig> {
+            panic!("just a mock impl to satisfy trait bounds")
+        }
+    }
+
+    #[tokio::test]
+    async fn wait_for_finalized_returns_err_when_usurped() {
+        let c = MockClient;
+        let stream_elements: Vec<SubstrateTxStatus<H256, H256>> = vec![
```

Comment:
you could also make a few  type aliases to make this code more readable such as:

```rust
type Hash = PolkadotConfig::Hash;
type TxProgress = crate::tx::TxProgress<PolkadotConfig, MockClient>;
type SubstrateTxStatus = crate::rpc::types::SubstrateTxStatus<Hash, Hash>>;
```